connectors/sources/servicenow.py

```diff
@@ -4,35 +4,58 @@
 # you may not use this file except in compliance with the Elastic License 2.0.
 #
 """ServiceNow source module responsible to fetch documents from ServiceNow."""
+import asyncio
 import json
 import os
-from functools import cached_property
+from functools import cached_property, partial
 
+import aiofiles
 import aiohttp
+import dateutil.parser as parser
+from aiofiles.os import remove
+from aiofiles.tempfile import NamedTemporaryFile
 from aiohttp.client_exceptions import ServerDisconnectedError
 
 from connectors.logger import logger
 from connectors.source import BaseDataSource, ConfigurableFieldValueError
-from connectors.utils import CancellableSleeps, iso_utc
+from connectors.utils import (
+    TIKA_SUPPORTED_FILETYPES,
+    CancellableSleeps,
+    ConcurrentTasks,
+    MemQueue,
+    convert_to_b64,
+    iso_utc,
+)
 
 RETRIES = 3
 RETRY_INTERVAL = 2
+QUEUE_MEM_SIZE = 25 * 1024 * 1024  # Size in Megabytes
+FILE_SIZE_LIMIT = 10485760
 FETCH_SIZE = 1000
 MAX_CONCURRENCY_SUPPORT = 10
 
+END_SIGNAL = "TASK_FINISHED"
 RUNNING_FTEST = (
     "RUNNING_FTEST" in os.environ
 )  # Flag to check if a connector is run for ftest or not.
 
+DEFAULT_QUERY = "ORDERBYsys_created_on^"
 ENDPOINTS = {
     "TABLE": "/api/now/table/{table}",
     "ATTACHMENT": "/api/now/attachment",
     "DOWNLOAD": "/api/now/attachment/{sys_id}/file",
 }
+DEFAULT_SERVICE_NAMES = (
+    "sys_user",
+    "sc_req_item",
+    "incident",
+    "kb_knowledge",
+    "change_request",
+)
 
 
 class InvalidResponse(Exception):
-    """Raise when retrieving invalid response..
+    """Raise when retrieving invalid response.
```

Comment:
```suggestion
```

nit: I don't think we need an explicit doc string here. `InvalidResponse` already tells enough IMO.